sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -619,11 +616,9 @@ case class DropTable(
 case class AlterTable(
     catalog: TableCatalog,
```

Comment:
@rdblue I'm ok to use `NamedRelation` as this is what it was before. But I want to point out that it's hard to completely hide the connector API like `TableCatalog` here.

Reply:
Right, but at least it keeps the read/write API separated. The catalog API and the read/write API are orthogonal to one another. We can replace the read/write API without replacing the catalog API, so no need to tie logical plans to the read/write API.